src/DocumentStore.zig

```diff
@@ -1087,6 +1083,24 @@ fn createDocument(self: *DocumentStore, uri: Uri, text: [:0]const u8, open: bool
 
     _ = handle.setOpen(open);
 
+    if (std.process.can_spawn and !self.bare_instance_no_server) {
+        const Server = @import("Server.zig");
+        const server = @fieldParentPtr(Server, "document_store", self);
+
+        server.job_queue_lock.lock();
+        defer server.job_queue_lock.unlock();
+
+        try server.job_queue.ensureUnusedCapacity(1);
+        server.job_queue.writeItemAssumeCapacity(.{
+            .find_build_file_for_uri = try server.allocator.dupe(u8, uri),
+        });
+    }
```

Comment:
You have also delayed the collection imports and cimports which can cause wrong results until `findBuildFileForUri` is done. 

The code that prints the `Opened document {s} with build file {s}` message should also be removed because there is no guarantee that at that point in time, the associated build file has been found.